### PR TITLE
New Package:py-ucsf-pyem

### PR DESCRIPTION
**Pull request by dorton21**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-ucsf-pyem/package.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    maintainers = ['dorton21']
+
+    # Using commit since releases haven't been updated in 2 years
+    version('2021-04-07', commit='ed0527f98657d21d887357426b74e5240d477fae')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-future@0.15:', type=('build', 'run'))
+    depends_on('py-numba@0.41:', type=('build', 'run'))
+    depends_on('py-seaborn@0.9:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ucsf-pyem/package.py:27]: Missing deps on numpy/scipy/matplotlib. Also, where do the version requirements come from?

File: var/spack/repos/builtin/packages/py-ucsf-pyem/package.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    depends_on('py-pandas@0.23.4:', type=('build', 'run'))
+    depends_on('py-pathos@0.2.1:', type=('build', 'run'))
+    depends_on('py-pyfftw@0.10:', type=('build', 'run'))
+    depends_on('py-healpy@1.11:', type=('build', 'run'))
+    depends_on('py-natsort@6.0:', type=('build', 'run'))
+
+    # Extracts files into bin for proper useage
+    @run_after('install')
+    def extraction(self):
+        install_tree('.', self.prefix.bin)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ucsf-pyem/package.py]: Doesn't this also install all source code to bin? I don't think we want that.

File: var/spack/repos/builtin/packages/py-ucsf-pyem/package.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    # Example useage
+    # "projection_subtraction.py --help"
+    def setup_run_environment(self, env):
+        env.prepend_path('PATH', self.prefix.bin)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ucsf-pyem/package.py]: Isn't this automatically added by Spack's prefix inspections?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-ucsf-pyem/package.py (hunk 1 above)
Comment (dorton21) [reply var/spack/repos/builtin/packages/py-ucsf-pyem/package.py:27]: https://github.com/asarnow/pyem/blob/master/requirements.txt

File: var/spack/repos/builtin/packages/py-ucsf-pyem/package.py (hunk 2 above)
Comment (dorton21) [reply var/spack/repos/builtin/packages/py-ucsf-pyem/package.py]: If I'm not mistaken this package is simply used via calling scripts like `star.py --copy-micrograph-coordinates original_particles.star from_cryosparc.star from_csparc_with_coords.star
` as seen here: https://github.com/asarnow/pyem/wiki/Export-from-cryoSPARC-v2
The default install doesn't give access to the scripts ergo a simple dump seems to work. Unless I'm misunderstanding how this package works altogether.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-ucsf-pyem/package.py]: What does the `python setup.py build` and `python setup.py install` steps do then? I wish they used entrypoints like normal Python packages os they would automatically get installed in `prefix.bin`.

Comment (dorton21) [reply var/spack/repos/builtin/packages/py-ucsf-pyem/package.py]: The default installation does not create a bin. The only files paths that are created are without the install_tree() dump are:
`py-ucsf-pyem-2021-04-07-.../lib/python3.8/site-packages/pyem/...` which does not contain all the correct scripts. And `py-ucsf-pyem-2021-04-07-.../lib/python3.8/site-packages/pyem-0.4-py3.8.egg-info/...` is also created which does not contain anything useful. Also a perl5 folder which is empty. I may be mistaken but I think the setup.py basically checks the required packages exist and that's about it.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-ucsf-pyem/package.py]: Would it be sufficient to do:
```suggestion
        mkdir(self.prefix.bin)
        install('*.py', self.prefix.bin)
```
or are there other files that are needed?

Comment (dorton21) [reply var/spack/repos/builtin/packages/py-ucsf-pyem/package.py]: After some testing, it looks like this suggestion will work. Good idea.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**